core/vm/evm.go

```diff
@@ -211,11 +197,18 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 
 	var creation bool
 	if !evm.StateDB.Exist(addr) {
-		if !isPrecompile && evm.chainRules.IsEIP158 && value.Sign() == 0 {
-			if evm.chainRules.IsPrague {
-				// proof of absence
-				tryConsumeGas(&gas, evm.Accesses.TouchAndChargeProofOfAbsence(addr.Bytes()))
+		if !isPrecompile && evm.chainRules.IsEIP4762 {
+			// add proof of absence to witness
+			wgas := evm.Accesses.TouchAndChargeProofOfAbsence(addr.Bytes())
+			if gas < wgas {
+				evm.StateDB.RevertToSnapshot(snapshot)
+				return nil, 0, ErrOutOfGas
 			}
+			gas -= wgas
+		}
```

Comment:
@jsign this is a potential issue here. I didn't remember why we would only add entries to the proof if the value was 0. Does that ring a bell?